### DISCUSSION TITLE
Trying to keyed model collections in mount()

### DISCUSSION THREAD

**Discussion by Rizky92** [livewire/livewire#5004] [1. Help]
I have a collection of models that would use its `id` for key, something like `Model::all()->keyBy('id')`. However, when mounting this, I found an issue that `->keyBy('id')` isn't applied and would use normal numbered keys (0, 1, 2...) instead. Any help would be appreciated.

Here are the codes:

AddLabelField.php
```
class AddLabelField extends Component
{
    public $attachedLabels;
    ...

    public function mount()
    {
        $this->attachedLabels = Label::all()->keyBy('id');
    }

    // some methods

    public function render()
    {
        return view('livewire.add-label-field')
            ->with('labels', Label::where('name', 'LIKE', "%{$this->labelInput}%")->get());
    }
}
```
add-label-field.blade.php
```
...
<div class="flex items-center gap-x-2 gap-y-1 mt-4">
    @foreach ($attachedLabels as $key => $attachedLabel)
        <x-admin.label
            :key="$attachedLabel->id"
            :label="$attachedLabel->name"
            wire:click.prevent="removeLabel({{ $attachedLabel->id }})" />
    @endforeach
</div>
...
```
debugbar result:
```
"data" => array:2 [
  "attachedLabels" => Illuminate\Database\Eloquent\Collection {#1644
    #items: array:3 [
      0 => App\Models\Label {#1728
        ...
      }
      1 => App\Models\Label {#1726
        ...
      }
      2 => App\Models\Label {#1725
        ...
      }
    ]
    #escapeWhenCastingToString: false
  }
  ...
]
```

**Comment by joshhanley** [comment]
@Rizky92 Livewire uses Laravel's queue serialiser under the hood, and it can serialise Eloquent Collections, but not grouped by collection.

So you could save the ungroup collection to a public property and then use computed properties to generate your grouped by collection. See https://laravel-livewire.com/docs/2.x/properties#computed-properties

Hope this helps!

**Reply by Rizky92** [reply]
@joshhanley hey there! Thanks for the reply. I have already resolved it by plucking the model before used then store the id afterwards. This allows me to get the id as key but I couldn't get additional attributes from the model. So I couldn't be sure to mark your reply as answer. I appreciate the helps.

**Reply by joshhanley** [reply]
@Rizky92 no worries! 🙂 any chance you could add your own answer below with the solution and mark it as answered? Thanks!
